algorithms/ar-bsrh/typescript/binary_search.ts

```diff
@@ -0,0 +1,23 @@
+function binarySearch(item_list, item) {
+    let first = 0;
+    let last = item_list.length - 1;
```

Comment:
Codacy found an issue: [Identifier 'item_list' is not in camel case.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)